apps/api/src/app/environments-v1/usecases/output-renderers/email-output-renderer.usecase.ts

```diff
@@ -96,7 +96,11 @@ export class EmailOutputRendererUsecase {
       const current = queue.shift()!;
 
       if (hasShow(current.node)) {
-        await this.handleShowNode(current.node, variables, current.parent);
+        const shouldShow = await this.handleShowNode(current.node, variables, current.parent);
+
+        if (!shouldShow) {
+          continue;
+        }
```

Comment:
There was a bug with `show` - if the `shouldShow = false` we already removed the entire node and we dont need to continue with that node further.